### PR TITLE
AMQP DT support for New Relic

### PR DESCRIPTION
**Pull request by iamemilio**

### CODE REVIEW

**Comment by iamemilio** [comment]
Ok, this is ready for a review. The AMQP tests will not pass because it requires a minimum version of the go agent in the go.mod that we have not yet released. It has to do this otherwise certain metric constants and attribute settings will not exist.

**Comment by nr-swilloughby** [comment]
Can you get the AMQP test to pass by temporarily adding a replace line in the go.mod to tell it to use the local code as the new version of the agent? Or is the issue deeper than that, and it really requires that the agent is released and identifies itself as the new version?

**Comment by iamemilio** [comment]
> Can you get the AMQP test to pass by temporarily adding a replace line in the go.mod to tell it to use the local code as the new version of the agent? Or is the issue deeper than that, and it really requires that the agent is released and identifies itself as the new version?

Yeah, I can do that if you want. We should remember to take that out before releasing though

**Comment by nr-swilloughby** [comment]
Otherwise, and based on your demo yesterday, it looks good to me.

**Review by nr-swilloughby — DISMISSED** [review]

**Comment by mirackara** [comment]
Besides the minor naming thing I mentioned this also looks good :)

**Comment by iamemilio** [comment]
> Besides the minor naming thing I mentioned this also looks good :)

what naming thing?

**Comment by mirackara** [comment]
My bad @iamemilio i forgot to publish the review :')

File: v3/newrelic/transaction.go [hunk 1]
```diff
@@ -588,6 +589,11 @@ func (tt TransportType) toString() string {
 	}
 }
 
+const (
+	RabbitMQMessageBrokerLibrary = "RabbitMQ"
```
[the remark this answers is not in the document]
Comment (iamemilio) [reply v3/newrelic/transaction.go:593]: ohh, yeah :)

[the remark this answers is not in the document]
Comment (iamemilio) [reply v3/newrelic/transaction.go:593]: I should remove these actually

**Review by mirackara — APPROVED** [review]

**Pull request merged**